python/sglang/srt/function_call/utils.py

```diff
@@ -50,3 +55,98 @@ def _is_complete_json(input_str: str) -> bool:
         return True
     except JSONDecodeError:
         return False
+
+
+def validate_tool_definitions(tools: List[Tool]) -> None:
+    """
+    Validate that tool definitions don't have conflicting $defs.
+
+    Raises:
+        ValueError: If there are conflicting tool definitions
+    """
+    all_defs = {}
+    for tool in tools:
+        if tool.function.parameters is None:
+            continue
+        # Make a copy to avoid modifying original
+        params = tool.function.parameters.copy()
+        defs = params.pop("$defs", {})
+        for def_name, def_schema in defs.items():
+            if def_name in all_defs and all_defs[def_name] != def_schema:
+                raise ValueError(
```

Comment:
Could we explore if there is a way to make this more concise? Right now this function is almost the same as `get_tool_schema_defs`. Just without ValueError raise.